auth/token_verifier.go

```diff
@@ -269,13 +267,13 @@ func (tv *tokenVerifier) verifyHeaderAndBody(token string) (*Token, error) {
 	}
 
 	issuer := tv.issuerPrefix + tv.projectID
-	if header.KeyID == "" {
+	if !isEmulator && header.KeyID == "" {
 		if payload.Audience == firebaseAudience {
 			return nil, fmt.Errorf("expected %s but got a custom token", tv.articledShortName)
 		}
 		return nil, fmt.Errorf("%s has no 'kid' header", tv.shortName)
 	}
-	if header.Algorithm != "RS256" {
+	if !isEmulator && header.Algorithm != "RS256" {
```

Comment:
This is different from [Node.js implementation](https://github.com/firebase/firebase-admin-node/blob/93362d5721fc45f3fc541543813d5739eda8ba2d/src/auth/token-verifier.ts#L200) but I believe this is OK because we can skip algorithm verification in emulator mode and the feature that can specify an algorithm for token verification is not required for this Go SDK.

Reply:
I think in emulator mode we should look for the algorithm `none`. @yuchenshi your thoughts?

Reply:
According to the design doc approved, the header should not be checked at all when `isEmulator` (to allow future changes). Therefore @maku693's implementation is correct and afaict is the same as the Node.js implementation. @maku693 would you mind explaining why you think it would be different from the Node.js implementation?

Reply:
@yuchenshi Ah, sorry about my confusing comment. I mean ”This is not exactly same as what Node.js SDK does”. Node.js implementation does checks algorithm, regardless of if it is in emulator mode or not, like @hiranya911 says.
So in emulator mode, this is basically same as Node.js implementation.

Reply:
Looks like  Node.js skips algorithm check in emulator mode now. So this is fine.

https://github.com/firebase/firebase-admin-node/blob/93362d5721fc45f3fc541543813d5739eda8ba2d/src/auth/token-verifier.ts#L200-L203